libraries/shared/fetcher/geth_rpc_storage_fetcher.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Vulcanize
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fetcher
+
+import (
+	"fmt"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/ethereum/go-ethereum/statediff"
+	"github.com/sirupsen/logrus"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/storage/utils"
+	"github.com/vulcanize/vulcanizedb/libraries/shared/streamer"
+)
+
+type GethRpcStorageFetcher struct {
+	statediffPayloadChan chan statediff.Payload
+	streamer             streamer.Streamer
+}
+
+func NewGethRpcStorageFetcher(streamer streamer.Streamer, statediffPayloadChan chan statediff.Payload) GethRpcStorageFetcher {
+	return GethRpcStorageFetcher{
+		statediffPayloadChan: statediffPayloadChan,
+		streamer:             streamer,
+	}
+}
+
+func (fetcher GethRpcStorageFetcher) FetchStorageDiffs(out chan<- utils.StorageDiff, errs chan<- error) {
+	ethStatediffPayloadChan := fetcher.statediffPayloadChan
+	clientSubscription, clientSubErr := fetcher.streamer.Stream(ethStatediffPayloadChan)
```

Comment:
Do we want to handle the `clientSubscription` errors? We could use a `select` to receive payloads from `ethStatediffPayloadChan` and errors from `clientSubscription.Err()` kinda like in the seed node [service](https://github.com/vulcanize/vulcanizedb/blob/ipfs_concurrency/pkg/ipfs/service.go#L149).

Reply:
Yeah, this is an interesting point. Originally I think I just had the client errors being send to the `errs` chan, but that ends up just warning the user that there was an error[ in the watcher.](https://github.com/vulcanize/vulcanizedb/blob/staging/libraries/shared/watcher/storage_watcher.go#L64) It seems like we may potentially want to stop the process (at least for now) if the subscription fails, so that we don't miss the failure, and able to manually intervene. 

But my thought is that long term, this would retry the subscription if it fails, or is dropped.

Reply:
@i-norden I updated it in bc794e4 - what do you think?

Reply:
bc794e4 LGTM! Looking closer at the subscription error, I realize now that it is already handled on [Geth's side ](https://github.com/vulcanize/go-ethereum/blob/statediffing/statediff/api.go#L74) since this is the same subscription object that is returned to the client... meaning there is actually a race condition if we try to handle the error on the client side with the current setup. This appears to be the [standard  pattern](https://github.com/vulcanize/go-ethereum/blob/statediffing/eth/filters/api.go#L158), but I'm unsure now how the client is supposed to be informed of a subscription error then... I guess the client is just dropped silently? But doesn't that leave the client listening on the dropped payload channel indefinitely? I feel like the error should be getting forwarded so that it can be received and handled by the client as well. Just thinking out loud at this point... let me know if you notice something.

Reply:
bc794e4 LGTM! Doesn't look like it included edits to this part though? I'm still not sure what the best way to handle the subscription errors is at this time, the original intention of that err channel is to signal resubscription like you intend https://github.com/vulcanize/go-ethereum/blob/statediffing/rpc/subscription.go#L229 but for now maybe we should just stop the process?